src/sentry/testutils/silo.py

```diff
@@ -283,6 +283,54 @@ def assume_test_silo_mode(desired_silo: SiloMode, can_be_monolith: bool = True)
         yield
 
 
+@contextmanager
+def assume_test_silo_mode_of(*models: Type[BaseModel], can_be_monolith: bool = True) -> Any:
+    """Potentially swap to the silo mode where the models are accessed.
+
+    The argument should be one or more model classes that are scoped to exactly one
+    non-monolith mode. That is, they must be tagged with `control_silo_only_model` or
+    `region_silo_only_model`. The enclosed context is swapped into the appropriate
+    mode, allowing the model to be accessed.
+
+    If no silo-scoped models are provided, no mode swap is performed.
+
+    The intent is that you list the cross-silo models that you intend to access
+    within the block. However, this is for the sake of expressiveness only. The
+    context will not actually check that you access only those models; it will allow
+    you to access any model that happens to share the same silo mode.
+    """
+    # Possible improvements for the future:
+    # 1. "Exempt" only the listed models; raise if other cross-silo models are accessed.
+    # 2. Check that every listed model is accessed; raise if unused ones are listed.
```

Comment:
I'm of two minds as to whether it's a good idea to do this "expressiveness only" approach that tolerates misleading arguments. On the one hand, it could be good to start using the more specific style as early as possible. On the other, it might be counterproductive to allow loose instances now, only to have to fix them when we start enforcing strictness later.

(On the third hand, I picked almost the worst possible time to attempt this idea, as we're just about done putting `assume_test_silo_mode` in every last place in the test suite. 🤦 Better late than never? 😅)

Reply:
> Exhaustively checking that all models were used doesn't seem that valuable to me, as these functions are meant as ergonomics for writing tests.

Yeah, I could have phrased that better. I'm not so worried about `assume_silo_mode_of(Foo, Bar)` where we actually only needed `Foo` but not `Bar`. But if it turns out to be *entirely* unneeded, that could be worth detecting.